controllers/promise_controller.go

```diff
@@ -498,8 +499,17 @@ func (r *PromiseReconciler) createResourcesForDynamicControllerIfTheyDontExist(c
 	err := r.Client.Create(ctx, &cr)
 	if err != nil {
 		if errors.IsAlreadyExists(err) {
-			// TODO: Handle updates of all Promise resources gracefully.
-			logger.Info("Cannot execute update on pre-existing ClusterRole")
+			existingCR := &rbacv1.ClusterRole{}
```

Comment:
should this use the controllerutil.CreateOrUpdate method instead of just create/capture error/update? 

https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#CreateOrUpdate